lib/Onyx.js

```diff
@@ -302,14 +351,43 @@ function keysChanged(collectionKey, partialCollection) {
                 const dataKeys = _.keys(partialCollection);
                 for (let j = 0; j < dataKeys.length; j++) {
                     const dataKey = dataKeys[j];
-                    subscriber.callback(cachedCollection[dataKey], dataKey);
+                    const newData = cachedCollection[dataKey];
+
+                    // If there is a selector, only trigger the callback when that particular data has changed
+                    if (subscriber.selector) {
+                        const newValue = getSimplifiedObject(newData, subscriber.selector);
+
+                        // If the selected value doesn't exist, exit early so that the callback doesn't get called at all
+                        if (_.isUndefined(newValue)) {
```

Comment:
Just thinking out loud - but is it possible that a selector can return `undefined` even though it had a value before? And if this is the case, then wouldn't we want to notify the callback that it had a value before, but is now `undefined`? 

Separately, why do we return early here, but not in other places where we read the value after running the selector?

Reply:
> Separately, why do we return early here, but not in other places where we read the value after running the selector?

This is a very specific case in the code. It's only when you are using `Onxy.connect()` and you're connected to an entire collection, and you're not using `waitForCollectionCallback`. For that case, the callback is expected to be triggered `n` times (n = number of items in the collection), and passed each item.

So, when using a selector, if nothing is returned (ie. you don't "care" about that data), then we need to skip triggering the callback for this loop iteration. Thus, an early return happens and we move onto the next item in the collection.

If the early return was not there, the callback would still be called `n` times, but sometimes it would be passed a value and sometimes it would be passed `undefined` which I didn't find as being very predictable behavior (it means you have to implement even more logic inside the callback to handle undefined data, which is weird to force onto the callback).